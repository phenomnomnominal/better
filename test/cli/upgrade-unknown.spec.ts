import { describe, it, expect } from 'vitest';

import { createFixture } from '../fixture.js';

const ARGV = ['node', './bin/betterer', 'upgrade'];

describe('betterer upgrade', () => {
  it(`should doesn't change things it doesn't need to change`, async () => {
    const { cliÎ” } = await import('@betterer/cli');

    const { cleanup, logs, paths } = await createFixture(
      'upgrade-unknown',
      {
        './.betterer.ts': `
const { BettererTest } = require('@betterer/betterer');
const { bigger } = require('@betterer/constraints');

let start = 0;

// Don't remove this important comment!
console.log('some stuff here!');

// This stuff shouldn't be in here but let's not break things:
module.exports = {
  num: 9,
  str: 'some string',
  array: [],
  set: new Set(),
  regexp: /regexp/,
  object: {
    notTest: () => 3
  }
}

module.exports.getsBetter = () => new BettererTest({
  test: () => start++,
  constraint: bigger
});

export function foo () {
  console.log('something');
}

function bar () {
  console.log('something');
}
        `
      },
      {
        logFilters: [/ðŸŒŸ Upgrading Betterer/]
      }
    );

    const fixturePath = paths.cwd;

    process.env.BETTERER_WORKER = 'false';

    await cliÎ”(fixturePath, ARGV);

    expect(logs).toMatchSnapshot();

    await cleanup();
  });
});
