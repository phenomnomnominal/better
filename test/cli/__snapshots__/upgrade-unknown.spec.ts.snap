// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`betterer upgrade should doesn't change things it doesn't need to change 1`] = `
Array [
  "ðŸŽ‰ Upgrading Betterer (0ms): 1 task done!
âœ… Upgrading \\"./.betterer.ts\\": done!
ãƒ» Nothing to upgrade!",
]
`;
